discord/voice_client.py

```diff
@@ -329,17 +332,27 @@ def is_connected(self):
     # audio related
 
     def _get_voice_packet(self, data):
-        buff = bytearray(len(data) + 12)
-        buff[0] = 0x80
-        buff[1] = 0x78
-
-        for i in range(0, len(data)):
-            buff[i + 12] = data[i]
-
-        struct.pack_into('>H', buff, 2, self.sequence)
-        struct.pack_into('>I', buff, 4, self.timestamp)
-        struct.pack_into('>I', buff, 8, self.ssrc)
-        return buff
+        header = bytearray(12)
+        nonce = bytearray(24)
+        box = nacl.secret.SecretBox(bytes(self.secret_key))
+        
+        # Formulate header
+        header[0] = 0x80
+        header[1] = 0x78
+        struct.pack_into('>H', header, 2, self.sequence)
+        struct.pack_into('>I', header, 4, self.timestamp)
+        struct.pack_into('>I', header, 8, self.ssrc)
+        
+        # Copy header to nonce's first 12 bytes
+        for i in range(0, len(header)):
```

Comment:
The libsodium libs require the nonce to be 24 bytes in length, wouldn't that make it 12 bytes?